client/src/modules/AdminCosting/Components/AdminCosting.tsx

```diff
@@ -0,0 +1,138 @@
+
+import axios from 'axios'
+
+import { useAuthMandatoryLogin } from '../../../auth/auth_utils'
+
+import '../Styles/Admin.Costing.css'
+
+/** Admin Costing Page
+ * Get costing estimates for GPT4 and GPT3.5 using data from courses that have
+ * at least an x amount of reviews.
+ */
+export const AdminCosting = () => {
+  const [minimumReviews, setMinimumReviews] = useState<number | "">("");
+  const [totalTokens, setTotalTokens] = useState<number>(0)
+  const [totalReviews, setTotalReviews] = useState<number>(0)
+  const [totalChars, setTotalChars] = useState<number>(0)
+  const [totalWords, setTotalWords] = useState<number>(0)
+  const [avgWords, setAvgWords] = useState<number>(0)
+  const [avgChars, setAvgChars] = useState<number>(0)
+  const [avgTokens, setAvgTokens] = useState<number>(0)
+  const [gpt4cost, setGpt4Cost] = useState<number>(0)
+  const [gpt3cost, setGpt3Cost] = useState<number>(0)
+
+  const { isLoggedIn, token, isAuthenticating } = useAuthMandatoryLogin('admin')
+  const [loading, setLoading] = useState(true)
+  const [isAdmin, setIsAdmin] = useState(false)
+
+  useEffect(() => {
+    async function confirmAdmin() {
+      const res = await axios.post(`/api/tokenIsAdmin`, {
+        token: token,
+      })
+
+      if (res.data.result) {
+        setIsAdmin(true)
+      } else {
+        setIsAdmin(false)
+      }
+      setLoading(false)
+    }
+
+    if (isLoggedIn) {
+      confirmAdmin()
+    }
+  }, [isLoggedIn, token, isAuthenticating])
+
+  // Takes in a minimum number of reviews selected from a dropdown menu.  
+  // Gets data from all courses that have at least that number of reviews.
+  function getCostingData(min: number, setTotalTokens: React.Dispatch<React.SetStateAction<number>>) {
+    console.log('loading costing data...');
+
+    axios.post('/ai/costing', { min })
+      .then((response) => {
+        const result = response.data.result;
+        console.log('Costing Data Loaded');
+        setTotalTokens(result.tokens);
+        setTotalChars(result.chars);
+        setTotalWords(result.words);
+        setTotalReviews(result.reviews);
+        setAvgTokens(result.avgtokens);
+        setAvgWords(result.avgwords);
+        setAvgChars(result.avgchar);
+        setGpt3Cost(result.gpt3cost);
+        setGpt4Cost(result.gpt4cost);
+      })
+      .catch((error) => {
+        console.error('Error loading costing data:', error);
+      });
+  }
+
+  function renderAdmin(token: string) {
+    return (
+      <div>
+        <div className="heading">
+          <h1>GPT Costing Page</h1>
+          <label htmlFor="minReviews">For courses with minimum of</label>
+          <select
+            id="minReviews"
+            value={minimumReviews}
+            onChange={(e) => {
+              const value = Number(e.target.value);
+              setMinimumReviews(value);
+              getCostingData(value, setTotalTokens);
+            }}
+          >
+            <option value="100000">x</option>
```

Comment:
Maybe instead of setting the initial or "empty" state as an arbitrarily high limit for minimum reviews, it could just default to the lowest minimum number of reviews (3).